web/modules/custom/joinup_core/joinup_core.behat.inc

```diff
@@ -79,4 +81,57 @@ class JoinupCoreSubContext extends DrupalSubContextBase implements DrupalSubCont
     $this->visitPath($term->toUrl()->toString());
   }
 
+  /**
+   * Clicks on a link matching a pattern given the email text and the recipient.
+   *
+   * Note that if multiple messages have been sent to the same user, all of them
+   * will be checked but only the first instance of the link matching the
+   * pattern will be clicked.
+   *
+   * @param string $pattern
+   *   The pattern that the link should match to.
+   * @param string $mail
+   *   The recipient's mail.
+   *
+   * @throws \Exception
+   *   If no mail was found linked to the given email or if the pattern did not
+   *   find any matches.
+   *
+   * @Given I click the link matching the :pattern pattern from the email sent to :mail
+   */
+  public function clickMailLinkMatchingPattern(string $pattern, string $mail): void {
```

Comment:
I would move this into `joinup_notification.behat.inc` because this depends on the `@email` tag that is declared in that file. Otherwise we introduce a dependency from `joinup_core.behat.inc` on `joinup_notification.behat.inc`.